src/writer/ai/__init__.py

```
@@ -1041,6 +1043,36 @@ def delete_file(
+
+
+def catch_guardrail_error(func: Callable[..., R]) -> Callable[..., R]:
+    @wraps(func)
+    def wrapper(*args: Any, **kwargs: Any) -> R:
+        try:
+            return func(*args, **kwargs)
+        except BadRequestError as e:
+            parsed = e.response.json()
+            errors = parsed.get("errors")
+            if not errors:
+                raise e
```

Ruff (0.14.6)

Use `raise` without specifying exception name

Remove exception name

(TRY201)